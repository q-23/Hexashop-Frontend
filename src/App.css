.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.fa-stack[data-count]:after{
  position:absolute;
  right:0%;
  top:1%;
  content: attr(data-count);
  font-size:30%;
  padding:.6em;
  border-radius:999px;
  line-height:.75em;
  color: white;
  background:rgba(255,0,0,.85);
  text-align:center;
  min-width:2em;
  font-weight:bold;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
