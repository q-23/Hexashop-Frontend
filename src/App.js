import React, {useEffect, useState} from 'react';
import {Switch, Route} from 'react-router-dom';

import FlexContainer from "./components/_Shared/FlexContainer";
import Container from "./components/_Shared/Container";
import WithLoader from "components/_HOC/WithLoader";
import Navbar from "./components/Navbar";
import Header from "./components/Header";
import Menu from "./components/Menu";

import { routes } from "./routes";

function App() {
  const [categoriesData, setCategoriesData] = useState([]);
  const ContainerWithLoader = WithLoader(FlexContainer);


  useEffect(() => {
    const fetchCategories = async () => {
      const categoriesBuffer = await fetch(`${process.env.REACT_APP_API_URL}/category`, {
        headers : {
          'Content-Type': 'application/json',
          'Accept': 'application/json'
        }
      });
      const categoriesResolved = await categoriesBuffer.json();
      setCategoriesData(categoriesResolved.map(e => ({ showInMenu: true, ...e })));
    }
    fetchCategories();
  }, [])

  return (
    <>
      <Navbar/>
      <Header/>
      <Menu categories={categoriesData}/>
      <Container main mx_auto>
        <ContainerWithLoader isLoading={false} main_container>
            <Switch>
              {
                routes.map((route, idx) =>
                  <Route exact={route.exact} path={route.path} component={route.component} key={route.path + idx}/>
                )
              }
            </Switch>
        </ContainerWithLoader>
      </Container>
    </>
  );
}

export default WithLoader(App);
