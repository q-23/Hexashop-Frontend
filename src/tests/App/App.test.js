import React from 'react';
import { render } from '@testing-library/react';
import App from 'App';
import { MemoryRouter as Router } from 'react-router-dom';
import StateProviderMenu from "contexts/reducers/menu";

test('renders learn react link', () => {
  const wrapper = render(
    <Router keyLength={0}>
      <StateProviderMenu>
        <App />
      </StateProviderMenu>
    </Router>
    );
  expect(wrapper).toMatchSnapshot();
});
