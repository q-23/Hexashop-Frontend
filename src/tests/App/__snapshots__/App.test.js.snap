// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-AxhUy hszLNi"
      >
        <button
          class="sc-AxgMl eaxotf"
        >
          <i
            class="sc-AxheI iYlCMY fa fa-bars"
            color="#FFF"
          />
        </button>
        <button
          class="sc-AxgMl eaxotf"
        >
          <i
            class="sc-AxheI iYlCMY fa fa-github"
            color="#FFF"
          />
        </button>
        <button
          class="sc-AxgMl cvWGsZ"
        >
          <i
            class="sc-AxheI iYlCMY fa fa-user"
            color="#FFF"
          />
        </button>
        <button
          class="sc-AxgMl gumKSq"
        >
          <i
            class="sc-AxheI iYlCMY fa fa-shopping-basket"
            color="#FFF"
          />
        </button>
      </nav>
      <header
        class="sc-AxmLO gvPdNN"
      >
        <div
          class="sc-AxjAm fMxdlc"
        >
          <div
            class="sc-Axmtr jFpjaR"
            width="md"
          >
            <img
              alt="logo"
              class="sc-fzozJi fjzWkt"
              src="logo.png"
            />
          </div>
        </div>
      </header>
      <nav
        class="sc-fzoLsD jCfqXS"
        width="md"
      >
        <ul />
      </nav>
      <div
        class="sc-fzpans jQlRmq"
      />
      <div
        class="sc-AxirZ fYKBAj"
      >
        <div
          class="sc-AxjAm cKqwrr"
        >
          <div
            class="sc-AxjAm bXpdfR"
          >
            <p
              class="sc-fznKkj cwPxaC"
            >
              Some text...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-AxhUy hszLNi"
    >
      <button
        class="sc-AxgMl eaxotf"
      >
        <i
          class="sc-AxheI iYlCMY fa fa-bars"
          color="#FFF"
        />
      </button>
      <button
        class="sc-AxgMl eaxotf"
      >
        <i
          class="sc-AxheI iYlCMY fa fa-github"
          color="#FFF"
        />
      </button>
      <button
        class="sc-AxgMl cvWGsZ"
      >
        <i
          class="sc-AxheI iYlCMY fa fa-user"
          color="#FFF"
        />
      </button>
      <button
        class="sc-AxgMl gumKSq"
      >
        <i
          class="sc-AxheI iYlCMY fa fa-shopping-basket"
          color="#FFF"
        />
      </button>
    </nav>
    <header
      class="sc-AxmLO gvPdNN"
    >
      <div
        class="sc-AxjAm fMxdlc"
      >
        <div
          class="sc-Axmtr jFpjaR"
          width="md"
        >
          <img
            alt="logo"
            class="sc-fzozJi fjzWkt"
            src="logo.png"
          />
        </div>
      </div>
    </header>
    <nav
      class="sc-fzoLsD jCfqXS"
      width="md"
    >
      <ul />
    </nav>
    <div
      class="sc-fzpans jQlRmq"
    />
    <div
      class="sc-AxirZ fYKBAj"
    >
      <div
        class="sc-AxjAm cKqwrr"
      >
        <div
          class="sc-AxjAm bXpdfR"
        >
          <p
            class="sc-fznKkj cwPxaC"
          >
            Some text...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
